tests/corrupted_stamp_test.cpp

```diff
@@ -0,0 +1,318 @@
+        {-EILSEQ, 1, 100, 101, DNET_EXT_VERSION_V1, DNET_SERVER_SEND_BUGFIX_TIMESTAMP, {0}, {0}},
+        {-EILSEQ, (1 << 9) - 1, 100, 101, DNET_EXT_VERSION_V1, DNET_SERVER_SEND_BUGFIX_TIMESTAMP, {0}, {0}},
+        {-EILSEQ, (1 << 9) - 1, 1, 1024, DNET_EXT_VERSION_V1, DNET_SERVER_SEND_BUGFIX_TIMESTAMP, {0}, {0}},
+
+        // Non-valid cases.
+        // flags not in range
+        {0, 0, 100, 100, DNET_EXT_VERSION_V1, 0, {0}, {0}},
+        {0, 1 << 9, 100, 100, DNET_EXT_VERSION_V1, 0, {0}, {0}},
+
+        // data_size <= disk_size
+        {0, 1, 100, 99, DNET_EXT_VERSION_V1, 0, {0}, {0}},
+        {0, 1 << 9, 100, 42, DNET_EXT_VERSION_V1, 0, {0}, {0}},
+
+        // disk_size == 0
+        {0, 1, 0, 99, DNET_EXT_VERSION_V1, 0, {0}, {0}},
+        // version != 1
+        {0, 1, 1, 99, 0, 0, {0}, {0}},
+        {0, 1, 1, 99, 2, 0, {0}, {0}},
+        // timestamp out of range.
+        {0, 1, 100, 101, DNET_EXT_VERSION_V1, DNET_SERVER_SEND_BUGFIX_TIMESTAMP + 1, {0}, {0}},
+        {0, 1, 100, 101, DNET_EXT_VERSION_V1, DNET_SERVER_SEND_BUGFIX_TIMESTAMP + 42, {0}, {0}},
+        {0, 1, 100, 101, DNET_EXT_VERSION_V1, DNET_SERVER_SEND_BUGFIX_TIMESTAMP + 100500, {0}, {0}},
+        // broken timestamp + some padding
+        {0, 1, 100, 101, DNET_EXT_VERSION_V1, DNET_SERVER_SEND_BUGFIX_TIMESTAMP + 1, {1, 0, 1}, {1, 1}},
+
+        // Broken on padding
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {1, 0, 0}, {0, 0}},
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {0, 1, 0}, {0, 0}},
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {0, 0, 1}, {0, 0}},
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {0, 0, 0}, {1, 0}},
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {0, 0, 0}, {0, 1}},
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {0, 0, 0}, {1, 1}},
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {1, 1, 1}, {0, 0}},
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {1, 0, 1}, {0, 0}},
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {1, 1, 1}, {1, 0}},
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {1, 1, 1}, {0, 1}},
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {1, 0, 1}, {1, 0}},
+        {0, 1, 42, 42, DNET_EXT_VERSION_V1, 0, {1, 0, 1}, {0, 1}}
+    };
+
+    std::vector<headers_test_case> functional_cases = {
+        {-EILSEQ, 1, 100, 100, DNET_EXT_VERSION_V1, 0, {0}, {0}},
+        {0, 1, 100, 99, DNET_EXT_VERSION_V1, 0, {0}, {0}},
+
+        {-EILSEQ, 1, 100, 100, DNET_EXT_VERSION_V1, 0, {0}, {0}},
+        {0, 1 << 9, 100, 42, DNET_EXT_VERSION_V1, 0, {0}, {0}},
+
+        {-EILSEQ, 1, 100, 100, DNET_EXT_VERSION_V1, DNET_SERVER_SEND_BUGFIX_TIMESTAMP, {0}, {0}},
+        // disk_size == 0
+        {0, 1, 0, 99, DNET_EXT_VERSION_V1, 0, {0}, {0}},
+        // Broken padding
+        {0, 1, 0, 99, DNET_EXT_VERSION_V1, 0, {0, 1, 0}, {0}},
+    };
+}
+
+namespace tests {
+
+namespace bu = boost::unit_test;
+
+const std::vector<int> groups{1};
+
+nodes_data::ptr configure_test_setup(const std::string &path) {
```

Comment:
Please, move it to the top of the file. We try to keep cluster initialization at the top.